compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupport.java

```diff
@@ -68,6 +68,24 @@ public static boolean isHexLiteral(String numericLiteral) {
         return false;
     }
 
+    /**
+     * Check input for decimal indicator.
+     *
+     * @param literalValue literal to check
+     * @return true if Hex prefix is present, false otherwise
+     */
+    public static boolean isHexIndicator(String literalValue) {
+        int length = literalValue.length();
+        // There should be at least 3 characters to form hex literal.
+        if (length < 3) {
+            return false;
+        }
+        char firstChar = literalValue.charAt(1);
+        char secondChar = literalValue.charAt(2);
+        return (firstChar == 'x' || secondChar == 'x' || firstChar == 'X' || secondChar == 'X');
```

Comment:
```suggestion
        return firstChar == 'x' || secondChar == 'x' || firstChar == 'X' || secondChar == 'X';
```

Shall we add a comment to say why we are checking both 2nd and 3rd chars?

Reply:
Fixed